src/main/java/seedu/ichifund/logic/commands/transaction/FilterTransactionCommand.java

```diff
@@ -32,16 +34,25 @@
             + PREFIX_YEAR + "2019 "
             + PREFIX_CATEGORY + "food ";
 
-    private final Predicate<Transaction> predicate;
+    private final Month month;
+    private final Year year;
+    private final Optional<Category> category;
 
-    public FilterTransactionCommand(Predicate<Transaction> predicate) {
-        this.predicate = predicate;
+    public FilterTransactionCommand(Month month, Year year, Optional<Category> category) {
+        this.month = month;
+        this.year = year;
+        this.category = category;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredTransactionList(predicate);
+        model.setTransactionContext(
```

Comment:
You might want to consider renaming these to fit the Builder pattern convention.

Reply:
Does not really fit the Builder pattern, but could modify code to fit it

Reply:
In a later issue I suppose, on low priority